other/pdb-maxunavailable-with-deployments/pdb-maxunavailable-with-deployments.yaml

```diff
@@ -0,0 +1,56 @@
+apiVersion: kyverno.io/v1
+kind: ClusterPolicy
+metadata:
+  name: pdb-maxunavailable-with-deployments
+  annotations:
+    policies.kyverno.io/title: PodDisruptionBudget maxUnavailable Non-Zero with Deployments
+    policies.kyverno.io/category: Other
+    kyverno.io/kyverno-version: 1.11.4
+    kyverno.io/kubernetes-version: "1.27"
+    policies.kyverno.io/subject: PodDisruptionBudget, Deployment
+    policies.kyverno.io/description: >-
+      A PodDisruptionBudget which sets its maxUnavailable value to zero prevents
+      all voluntary evictions including Node drains which may impact maintenance tasks.
+      This may be acceptable if there are no matching controllers, but if there are then
+      creation of such a PDB could allow unintended disruption. This policy enforces that
+      a PodDisruptionBudget may not specify the maxUnavailable field as zero if there are
+      any existing matching Deployments having greater than zero replicas.
+spec:
+  validationFailureAction: Audit
+  background: false
+  rules:
+  - name: pdb-maxunavailable
+    match:
+      any:
+      - resources:
+          kinds:
+          - PodDisruptionBudget
+          operations:
+          - CREATE
+          - UPDATE
+    context:
+    - name: deploymentreplicas
```

Comment:
If possible, I would suggest covering statefulsets as well with this policy.

Reply:
It could be done but it would have to be reworked with more context variables so the correct API call is made. I'll take this suggestion as a future enhancement request :)